settings/src/main/scala/org/typelevel/sbt/TypelevelSettingsPlugin.scala

```diff
@@ -200,7 +204,42 @@ object TypelevelSettingsPlugin extends AutoPlugin {
         Seq("-Werror")
       else
         Seq.empty
+    },
+    scalacOptions ++= {
+      val releaseOption = tlJdkRelease
+        .value
+        .map { release => if (isJava8) Seq.empty else Seq("-release", release.toString) }
```

Comment:
Why do we need the `isJava8` check for `scalacOptions`?

Also, what is the failure mode if `tlJdkRelease > 8` but `java.version == 1.8`?

Reply:
If `java.version == 1.8` it means we're running `javac 1.8` which doesn't understand the `--release` flag. `scalac` somehow makes use of this and also fails compilation with
```
[error] '8' is not a valid choice for '-target'
[21](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:21)
[error] '8' is not a valid choice for '-target'
[22](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:22)
[error] bad option: '-target:8'
[23](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:23)
[error] bad option: '-target:8'
[24](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:24)
[error] (kernelJVM / Compile / compileIncremental) Compilation failed
[25](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:25)
[error] (kernelJS / Compile / compileIncremental) Compilation failed
[26](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:26)
[error] Total time: 13 s, completed Feb 16, 2022 11:57:41 AM
[27](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:27)

[28](https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true#step:12:28)
Error: Process completed with exit code 1.
```

from https://github.com/typelevel/cats-effect/runs/5215692131?check_suite_focus=true

Reply:
Oh, cool, that seems good. But if we don't add the flag, it won't protected us from a target JDK newer the the compile-time JDK since it basically ignores that setting, so it seems we need another check?

Reply:
JDK 8 cannot target a newer JDK. Am I understanding your question correctly?

Reply:
If you don't specify anything, `javac 1.8` compiles JVM 8 bytecode, 17 compiles JVM 17 bytecode, while scalac compiles JVM 8 bytecode.

Reply:
Apologies if I'm not reading your code correctly. Here's my hypothetical situation:
1. `tlJdkRelease := Some(17)`
2. I load my project in sbt on Java 8
3. No release flags are added to `scalacOptions`, b/c `java.version == 1.8`
4. My project compiles, even though I asked to target JDK 17.

Reply:
If a project is specifically targetting 17 and the code is compilable by JDK 8, I don't know why they're targetting 17 in the first place. 😆

Reply:
That's a fair point :)

`scalac` can generate more optimized bytecode in certain situations when targeting newer JDKs.

Also, the fact that your project would fail to compile if you load it in Java 9 (but 8 would work!) is a bit odd imo :)

Reply:
e.g.
- https://github.com/lampepfl/dotty/pull/12929